src/archivematica/storage_service/common/backends.py

```diff
@@ -95,17 +115,53 @@ def decode_token(token: str) -> Any:
         return info
 
     def create_user(self, user_info: dict[str, Any]) -> User:
+        """Create a new user when authentication was successful."""
+        role = self.get_user_role(user_info)
+        if role is None:
+            return None
+
         user = super().create_user(user_info)
         for attr, value in user_info.items():
             setattr(user, attr, value)
-        self.set_user_role(user)
+        user.set_role(role)
         return user
 
     def update_user(self, user: User, user_info: dict[str, Any]) -> User:
-        self.set_user_role(user)
+        """Updates the user's role only if the setting allows roles to be set from OIDC claims."""
+        if self.OIDC_OP_SET_ROLES_FROM_CLAIMS:
+            role = self.get_user_role(user_info)
+            if role is None:
+                return None
+            user.set_role(role)
```

Comment:
Why is setting the role of the user now dependent on the `OIDC_OP_SET_ROLES_FROM_CLAIMS` setting?

> Added new OIDC setting that will allow admins to configure whether user
roles will be set from the OIDC claims, or assigned the system default
role. 

If I understand the new logic of the method correctly if the setting is disabled this will not set the system default role?

Also checking for the setting here and then calling `self.get_user_role` which also checks the setting seems unnecessary.